src/ext/php5/serializer.c

```diff
@@ -257,18 +257,79 @@ static void _serialize_exception(zval *el, zval *meta, ddtrace_exception_t *exce
         return;
     }
 
-    int needs_copied = zend_get_object_classname(exception, &class_name, &class_name_len TSRMLS_CC);
-
     add_assoc_long(el, "error", 1);
 
     zend_call_method_with_0_params(&exception, Z_OBJCE_P(exception), NULL, "getmessage", &msg);
-
-    /* add_assoc_stringl does not actually mutate the string, but we've either
-     * already made a copy, or it will when it duplicates with dup param, so
-     * if it did it should still be safe. */
-    add_assoc_stringl(meta, "error.type", (char *)class_name, class_name_len, needs_copied);
     add_assoc_zval(meta, "error.msg", msg);
 
+    bool use_class_name_for_error_type = true;
+    if (instanceof_function(Z_OBJCE_P(exception), ddtrace_ce_fatal_error TSRMLS_CC)) {
+        zval *code = NULL;
+        zend_call_method_with_0_params(&exception, Z_OBJCE_P(exception), NULL, "getcode", &code);
+        if (Z_TYPE_P(code) == IS_LONG) {
+            ddtrace_string error_type;
+            switch (Z_LVAL_P(code)) {
+                case E_ERROR:
+                    error_type = DDTRACE_STRING_LITERAL("E_ERROR");
+                    break;
+                case E_WARNING:
+                    error_type = DDTRACE_STRING_LITERAL("E_WARNING");
+                    break;
+                case E_PARSE:
+                    error_type = DDTRACE_STRING_LITERAL("E_PARSE");
+                    break;
+                case E_NOTICE:
```

Comment:
If `DDTrace\FatalError` is only created for `E_ERROR`, `E_CORE_ERROR`, and `E_USER_ERROR`, do we need all the other error codes here?

Reply:
I suppose not; was just trying to be robust and helpful. For instance, we use `"{unknown error}"` by default if it doesn't match anything, but it's a bit disingenuous if it's `E_WARNING`, `E_NOTICE`, etc.

Reply:
I like the idea of being helpful, but to me it's a bit confusing to support non-fatal errors when serializing `DDTrace\FatalError`. Perhaps a `ZEND_ASSERT(error_type == ...)` would help communicate the specific fatal errors that are supported here?

Reply:
Let me think about it a bit more. Assertion failures are hard failures but they are only present in debug builds.

Reply:
Yes, a hard failure would be ideal in this case IMO. If an `E_NOTICE` made its way into a `DDTrace\FatalError`, something is really broken. ;)

Reply:
I think the best way to do this is to drop the ones that aren't fatal, then doing something in the default branch. What do you think?

Reply:
Perfect 👍